src/foam/nanos/auth/CachingAuthService.java

```diff
@@ -51,10 +53,10 @@ public Object create(X x) {
   protected static Map<String,Boolean> getPermissionMap(final X x) {
     Session             session = (Session) x.get(Session.class);
     Map<String,Boolean> map     = (Map) session.getContext().get(CACHE_KEY);
+    User                user    = (User) x.get("user");
```

Comment:
This can be moved inside the `if` since we don't use it unless `map` is null. Saves a context lookup for every call to `check` when the cache is already there.

Reply:
agreed